src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -82,23 +82,33 @@ public void consumeAndNotify() {
         }
 
         while (!Thread.currentThread().isInterrupted()){
-            log.info("Waiting requests from publisher [queueSize:{}]", queue.size());
-            try {
-                String chainTaskId = queue.take();
-                taskUpdateLocks.runAsyncWithLock(chainTaskId, () -> consumer.onTaskUpdateRequest(chainTaskId));
-            } catch (InterruptedException e) {
-                log.error("The unexpected happened", e);
-                Thread.currentThread().interrupt();
-            }
+            waitForTaskUpdateRequest();
         }
     }
 
     /**
-     * Clear released locks every hour so that we don't have memory leaks.
+     * Wait for new task update request and run the update once a request is arrived.
+     * 2 requests can be run in parallel if they don't target the same task.
+     * <br>
+     * Interrupts current thread if queue taking has been stopped during the execution.
      */
-    @Scheduled(fixedDelay = 1000*60*60)
-    public void clearLocks() {
-        taskUpdateLocks.clearReleasedLocks();
+    private void waitForTaskUpdateRequest() {
+        log.info("Waiting requests from publisher [queueSize:{}]", queue.size());
+        try {
+            String chainTaskId = queue.take();
+            locks.putIfAbsent(chainTaskId, new AtomicBoolean(true)); // create lock if necessary
```

Comment:
Why do we need the `AtomicBoolean` here ? Maybe a simple `new Object()`.

Reply:
You're right! I don't think there any reason to keep an `AtomicBoolean` there. Let's keep it simple.